app/src/main/java/com/andrew/tasky/presentation/agenda_item_detail/AgendaItemDetailFragment.kt

```diff
@@ -89,32 +92,56 @@ class AgendaItemDetailFragment: Fragment(R.layout.fragment_agenda_item_detail) {
         }
     }
 
+    private fun setupAgendaInitialViewModels() {
+        if (!viewModel.isInitiallySetup.value) {
+            args.agendaItem?.let {
+                viewModel.setTitle(it.title)
+                viewModel.setDescription(it.description)
+                viewModel.setSelectedDate(it.startDateAndTime.toLocalDate())
+                viewModel.setSelectedTime(it.startDateAndTime.toLocalTime())
+                viewModel.setIsDone(it.isDone)
+                viewModel.setSelectedReminderTime(it.reminderTime)
+            }
+            viewModel.setEditMode(args.isInEditMode)
+            viewModel.setInitialSetupToTrue()
+        }
```

Comment:
All this is ViewModel specific code and can be put there. To access navigation arguments in your ViewModel, you can do this:
```
class YourViewModel(private val savedStateHandle: SavedStateHandle): ViewModel() { 

    private val agendaItem: AgendaItem? = savedStateHandle.get("agendaItem", null)
 }
```

and your ViewModel then needs to be initialized in your Fragment like this:
`private val viewModel: YourViewModel by viewModels()`

then you can also get rid of this line in your code:
`viewModel = ViewModelProvider(this)[AgendaItemDetailViewModel::class.java]`